Sources/BSON/ExtendedJSONEncoder.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+/// Facilitates the encoding of `Encodable` values into ExtendedJSON.
+public class ExtendedJSONEncoder {
+    /// The enum representing the two possible encoding modes.
+    public enum extendedJSONMode {
```

Comment:
I think we could just name this `Mode`, making the full type name `ExtendedJSONEncoder.Mode`.

Reply:
done